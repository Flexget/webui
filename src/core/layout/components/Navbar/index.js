import { connect } from 'react-redux';
import { LOGOUT } from 'core/auth/data/actions';
import { SERVER_RELOAD, SERVER_SHUTDOWN } from 'core/server/data/actions';
import { request } from 'utils/actions';
import Navbar from './Navbar';

function mapStateToProps({ router }) {
  return {
    pathname: router.location.pathname,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(request(LOGOUT)),
    reloadServer: () => dispatch(request(SERVER_RELOAD)),
    shutdownServer: () => dispatch(request(SERVER_SHUTDOWN)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
export { Navbar };
