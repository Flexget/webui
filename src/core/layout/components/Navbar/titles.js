export default {
  '/config': 'Config',
  '/history': 'History',
  '/log': 'Log',
  '/movies': 'Movies',
  '/pending': 'Pending',
  '/seen': 'Seen',
  '/series': 'Series',
  '/tasks/execute': 'Execute',
  '/tasks/schedule': 'Schedule',
  '/tasks/status': 'Status',
};
