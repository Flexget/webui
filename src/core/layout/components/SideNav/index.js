import { connect } from 'react-redux';
import SideNav from './SideNav';

function mapStateToProps({ routes }) {
  return {
    routes: Object.values(routes),
  };
}

export default connect(mapStateToProps)(SideNav);
