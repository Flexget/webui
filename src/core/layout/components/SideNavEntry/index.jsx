import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ListItemIcon } from 'material-ui/List';
import {
  SideNavIcon,
  SideNavText,
  NavItem,
} from './styles';

const SideNavEntry = ({ onClick, path, icon, name, className }) => {
  const item = (
    <NavItem className={className} onClick={onClick}>
      <ListItemIcon><SideNavIcon icon={icon} fixedWidth /></ListItemIcon>
      <SideNavText disableTypography inset primary={name} />
    </NavItem>
  );

  if (path) {
    return <Link to={path}>{item}</Link>;
  }

  return item;
};

SideNavEntry.propTypes = {
  onClick: PropTypes.func.isRequired,
  path: PropTypes.string,
  icon: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
};

SideNavEntry.defaultProps = {
  className: '',
  path: '',
};

export default SideNavEntry;
