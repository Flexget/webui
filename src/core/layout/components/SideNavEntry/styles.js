import styled, { css } from 'react-emotion';
import theme from 'theme';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

const colorClass = css`color: ${theme.palette.secondary[200]};`;

export const SideNavIcon = styled(FontAwesomeIcon)`
  ${colorClass};
`;

export const SideNavText = styled(ListItemText)`
  ${colorClass};
`;

export const NavItem = styled(ListItem)`
  border-left: 3px solid transparent;
  cursor: pointer;

  &:hover {
    border-left: 3px solid ${theme.palette.primary[500]};
  }
`;
