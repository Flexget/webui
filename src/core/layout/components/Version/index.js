import { connect } from 'react-redux';
import { GET_VERSION } from 'core/version/data/actions';
import { request } from 'utils/actions';
import Version from './Version';

export function mapStateToProps({ version }) {
  return {
    version,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getVersion: () => dispatch(request(GET_VERSION)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Version);
export { Version };
