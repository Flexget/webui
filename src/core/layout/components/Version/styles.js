import styled from 'react-emotion';
import theme from 'theme';

export const Wrapper = styled.div`
  color: ${theme.palette.error[500]};
`;

export const Line = styled.p`
  margin: 0;
`;
