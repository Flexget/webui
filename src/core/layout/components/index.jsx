import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'core/layout/components/Logo';
import Navbar from 'core/layout/components/Navbar';
import SideNav from 'core/layout/components/SideNav';
import LoadingBar from 'components/LoadingBar';
import ErrorStatus from 'components/ErrorStatus';
import InfoStatus from 'components/InfoStatus';
import {
  Wrapper,
  Main,
  Header,
  LogoWrapper,
  Nav,
  SideBar,
  Content,
} from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState(({ sideBarOpen }) => ({
      sideBarOpen: !sideBarOpen,
    }));
  }

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav
              sideBarOpen={sideBarOpen}
              toggle={this.toggleSideBar}
            />
          </SideBar>
          <Content open={sideBarOpen}>
            { children }
          </Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
