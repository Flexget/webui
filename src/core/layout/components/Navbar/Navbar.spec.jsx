import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Navbar } from 'core/layout/components/Navbar';
import { themed, router } from 'utils/tests';
import { NavIcon } from './styles';

function renderNavbar(toggle, logout) {
  return router(themed(<Navbar
    toggle={toggle}
    logout={logout}
    reloadServer={jest.fn()}
    shutdownServer={jest.fn()}
    classes={{}}
    titleMap={{}}
  />));
}

describe('core/layout/components/Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      renderNavbar(jest.fn(), jest.fn())
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('calls toggle when the hamburger button is pressed', () => {
    const toggle = jest.fn();
    const logout = jest.fn();
    const wrapper = shallow(<Navbar
      toggle={toggle}
      logout={logout}
      classes={{}}
      reloadServer={jest.fn()}
      shutdownServer={jest.fn()}
      titleMap={{}}
    />);

    wrapper.find('.fa-bars').closest(NavIcon).simulate('click');
    expect(toggle).toHaveBeenCalled();
  });

  xit('calls handleMenuClick when the menu button is pressed', () => {
    const toggle = jest.fn();
    const logout = jest.fn();
    const target = { a: 'target' };
    const wrapper = shallow(<Navbar
      toggle={toggle}
      logout={logout}
      classes={{}}
      reloadServer={jest.fn()}
      shutdownServer={jest.fn()}
      titleMap={{}}
    />);

    const button = wrapper.find('.fa-cog').closest(NavIcon);

    button.simulate('click', { currentTarget: target });
    expect(wrapper.state().menuOpen).toEqual(true);
    expect(wrapper.state().anchorEl).toEqual(target);
  });
});
