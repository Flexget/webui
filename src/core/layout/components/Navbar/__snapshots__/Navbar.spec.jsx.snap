// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/layout/components/Navbar renders correctly 1`] = `
.emotion-12 {
  position: static;
}

.emotion-10 {
  background-color: #ef6c00;
  color: #fff;
  min-height: 5rem;
}

.emotion-0 {
  color: #fff;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed emotion-12 emotion-13 MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular emotion-10 emotion-11 MuiToolbar-gutters"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root emotion-0 emotion-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      />
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
    />
    <div
      className="emotion-2 emotion-3"
    />
    <a
      href="/config"
      onClick={[Function]}
    >
      <button
        aria-label="Config"
        className="MuiButtonBase-root MuiIconButton-root emotion-0 emotion-1"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        />
      </button>
    </a>
    <a
      href="/log"
      onClick={[Function]}
    >
      <button
        aria-label="Log"
        className="MuiButtonBase-root MuiIconButton-root emotion-0 emotion-1"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        />
      </button>
    </a>
    <button
      aria-label="Manage"
      className="MuiButtonBase-root MuiIconButton-root emotion-0 emotion-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      />
    </button>
  </div>
</header>
`;
